src/sys/unix/selector/adapters/level_triggered/io_source_state.rs

```diff
@@ -0,0 +1,106 @@
+use crate::sys::Selector;
+use crate::{Interest, Registry, Token};
+use std::io;
+use std::os::unix::io::RawFd;
+use std::sync::Arc;
+use crate::sys::unix::selector::poll::RegistrationRecord;
+
+struct InternalState {
```

Comment:
This is a lot of state per fd.

The wasi implementation is based of Posix `poll` (Wasi's `poll_oneoff`) and doesn't need to hold any state, but requires proper (de)registering, can we follow that implementation more closely?

Reply:
This is basically a copy of how the Windows impl does it, and was the same in #1602.  I don't think the wasi approach will work, and actually I'm not really sure the wasi implementation is event correct.  It's certainly missing a lot of features and doesn't behave the same as epoll/kqueue.  The poll behaviour in this PR is trying to be as close as possible to a correct epoll emulation with only one leak that I'm aware of (see the change to tcp_stream test)